rfc/0037-quilt-mappings-structure.md

```diff
@@ -0,0 +1,51 @@
+The `main` branch will be for development only, meaning no commits happen on the release branches without happening on the `main` branch first. 
+
+This RFC also outlines how pull requests are accepted.
+
+## Motivation
+
+Some motivation behind this change is that the snapshot phases for mappings have often introduced refactors, which if not back ported, can cause quite a bit of pain for modders updating to the latest version, as the `migrateMappings` task is not used as much as it should. 
+The first release should help with this, as they will only have to deal with changed/removed methods, and not new names. 
+In addition, the build number for mappings can often jump from `1` to `15` overnight, then stagnate for a long time. 
+
+The motivation behind a single development branch stems from the fact that Yarn effectively works like this already, as most versions have their branch start from the end of the previous version's. 
+The only downside to this change is that it could be harder to find the mappings for a snapshot, but this could be solved with tags on new snapshots.
+The snapshot branch would allow for many new changes to be merged in at once, then be ported as a single new version to the release branch. This also removes new releases that happen with no actual changes to the mappings.
+
+For refactors and back ports, I believe a main development branch will make things easier, as the changes to the current Minecraft version can be cherry picked back 90% of the time.
+
+## Explanation
+
+### Branches
+
+* Branches for every major release version of Minecraft, with the initial commit be just the changes from the previous major version.
+* A single branch for development.
+
+The releases branch would publish to the releases maven, while snapshots would be published to the snapshot maven. This would help people scanning the maven find the latest version easier.
+
+### Versioning
+
+With the development branch and release branches, it might be possible to have consistent versioning across Minecraft versions, so the gradle dependency name could be `org.quiltmc:quilt-mappings:${mappings-version}-${minecraft-version}`
+
+### Pull Requests
+
+New mappings will always happen to the `main` branch, and should be reviewed by at least 3 members of the mappings team. (This number can change depending on how many people are on the teams, this is the number Fabric used).
+Pull requests should have the `snapshot` or `refactor` target. 
+* The `snapshot` pull requests should only happen during snapshot phases of Minecraft's development, and will only change mappings introduced between major versions.
+* The `refactor` pull requests can happen at any time, and should only update mappings from the current and previous versions of Minecraft. This will help increase consistency between names across versions. If a refactor wants to change names that follow the new changes, but are in only one of the versions, another pull request should accompany it.
```

Comment:
If some mappings get out of date during snapshots, will you have to create two pull requests now (one for new mappings, and one for changing the outdated ones)?

Reply:
say `m_asdf` is in 1.16.5 and `m_fdsa` is in 1.17. they are both named `badName`. someone wants to refactor to `goodName`. is this what you are asking?

Reply:
two pull requests would be needed, but also know that with hashed mojmap any mojang rename or added parameters would loose the mappings, which is somewhat intended. However, I should add a section about pulling mappings with hashed changes or parameter changes to the same method

Reply:
> know that with hashed mojmap any mojang rename or added parameters would loose the mappings

Ah, right. Forgot about that.

An example would be `MixedNoisePoint`, which kept its name in 1.18, but isn't actually a point anymore. So if someone would create a PR with 1.18 mappings, they would have to open a separate PR for this refactor.

But with hashed Mojmap (and small refactor PRs probably getting merged rather quickly), this shouldn't be a problem.

Reply:
So correction, adding parameters does not lose the name, so only one pull request would be needed in the example above.

I think I understand what you are saying about opening a separate PR, and it would not necessarily be a refactor as the experimental snapshots arent part of the normal game update path.